### PR TITLE
Client 2836 pkg

### PR DESCRIPTION
**Pull request by vmsachin**
[Added rpm packaging, Added deb packaging and Added git actions workflow to automate pkg creation]

### CODE REVIEW

File: .github/workflows/build.yml [hunk 1]
```diff
@@ -1,51 +1,96 @@
...
-        run: |
-          curl --proto '=https' --tlsv1.2 -sSf https://sh.rustup.rs | sh -s -- -y
-          source $HOME/.cargo/env
-
-      - name: Install PHP Build Dependencies
-        run: |
-          sudo apt-get update
-          sudo apt-get install -y autoconf2.13 bison2.7 libbison-dev libcurl4-openssl-dev libssl-dev libxml2-dev libjpeg-dev libpng-dev libicu-dev libmcrypt-dev libreadline-dev libfreetype6-dev libzip-dev
+    build-php-client:
+        runs-on: ubuntu-latest
```
Comment (khaf) [inline .github/workflows/build.yml:6]: No RPM yet?

File: pkg/rpm/scripts/postinst [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+GO_VERSION=$(go version | grep -oP '\d+\.\d+\.\d+')
+REQUIRED_GO_VERSION="1.20.0"
+
+if ! dpkg --compare-versions "$GO_VERSION" ">=" "$REQUIRED_GO_VERSION"; then
+    log_error_and_exit "Go version $REQUIRED_GO_VERSION or above is required."
+fi
+
+
+# Set environment variable for binary name
+export AS_CONNECTION_MANAGER="asld"
```
Comment (khaf) [inline pkg/rpm/scripts/postinst:64]: `asld`? IS the name correct?

**Review by khaf — APPROVED** [review]
Added a few comments. Seems ok to merge.

File: .github/workflows/build.yml (hunk 1 above)
Comment (vmsachin) [reply .github/workflows/build.yml:6]: not yet, will add it by the next patch.

File: pkg/rpm/scripts/postinst (hunk 2 above)
Comment (vmsachin) [reply pkg/rpm/scripts/postinst:64]: yes, we are using this everywhere in package as well. It follows convention to keep binary name small.

**Pull request merged**